flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -420,6 +421,8 @@ private String getShardIterator(GetShardIteratorRequest getShardIteratorRequest)
     protected boolean isRecoverableSdkClientException(SdkClientException ex) {
         if (ex instanceof AmazonServiceException) {
             return KinesisProxy.isRecoverableException((AmazonServiceException) ex);
+        } else if (ex.getCause() instanceof ConnectTimeoutException) {
```

Comment:
nit: You could use `ExceptionUtils#hasCause` incase the exception gets wrapped:
- https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/exception/ExceptionUtils.html#hasCause-java.lang.Throwable-java.lang.Class-

Reply:
How about having the test be`ExceptionUtils.findThrowable(ex, ConnectException.class).isPresent()`? Using `java.net.ConnectException` has the advantage that it is independent of the HTTP library used by the AWS SDK. (The `ExceptionUtils` here is from `org.apache.flink.util`).

Or, using `ExceptionUtils` from `org.apache.commons.lang3`, the test could be `ExceptionUtils.getRootCause(ex) instanceof ConnectException`.

Reply:
Yes I agree, either approach is fine. I will wait for an update and merge once tests pass

Reply:
Updated, tests pass.